resources/design.doc2.md

```diff
@@ -0,0 +1,181 @@
+# Design Document
+
+## _Project Title_ Design
+
+## 1. Problem Statement
+
+_The objective of this project is to create a web application using Java, that helps streamline collaboration among team members working on projects together. The app will allow managers to create projects and assign tasks, while team members can track, prioritize and update progress of their assigned tasks, resulting in an efficient tool that increases project transparency for teams while working together._
+
+## 2. Use Cases
+
+U1. As a manager, I want to create a new, empty project with a given name.
+
+U2. As a manager, I want to create a new, empty task with a given task name, description, due date, and list of assigned team members.
+
+U3. As a manager, I want to add a task, to a project with a given name.
+
+U4. As a manager, I want to update a task with different information.
+
+U5. As a manager, I want to delete a task with a given ID.
+
+U6. As a manager, I want to view all projects I have created.
+
+U7. As a team member, I want to retrieve all tasks assigned to me.
+
+U8. As a team member, I want to retrieve all tasks assigned to me in a prioritized order (due date).
+
+U9. As a team member, I want to be able to update the progress of specific task (Not Started, In Progress, Complete).
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+## 3. Stretch Goals
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+U11. As a team member, I want to be able to comment on a specific task I am assigned to.
+
+U13. As a manager, I want to be able to share and collaborate files and documents with my team.
+
+## 4. Project Scope
+
+### 4.1. In Scope
+
+_Which parts of the problem defined in Sections 1 and 2 will you solve with this design? This should include the base functionality of your product. What pieces are required for your product to work?_
+
+_The functionality described above should be what your design is focused on. You do not need to include the design for any out of scope features or expansions._
+
+### 4.2. Out of Scope
+
+_Based on your problem description in Sections 1 and 2, are there any aspects you are not planning to solve? Do potential expansions or related problems occur to you that you want to explicitly say you are not worrying about now? Feel free to put anything here that you think your team can't accomplish in the unit, but would love to do with more time._
+
+_The functionality here does not need to be accounted for in your design._
+
+# 5. Proposed Architecture Overview
```

Comment:
Please render an image of your UML diagrams (class and sequence) and embed them in this section so that someone can read them without having to render the plantuml.